tests/TokenizerTest.php

```diff
@@ -7,9 +7,36 @@
 use Doctrine\SqlFormatter\Tokenizer;
 use PHPUnit\Framework\Attributes\DoesNotPerformAssertions;
 use PHPUnit\Framework\TestCase;
+use ReflectionClass;
+
+use function array_diff;
+use function explode;
 
 final class TokenizerTest extends TestCase
 {
+    public function testKeywordsReservedToplevelAndNewlineAreReserved(): void
+    {
+        $tokenizerReflClass = new ReflectionClass(Tokenizer::class);
+        /** @var list<string> $tokenizerReserved */
+        $tokenizerReserved = $tokenizerReflClass->getProperty('reserved')->getDefaultValue();
+        /** @var list<string> $tokenizerReservedToplevel */
+        $tokenizerReservedToplevel = $tokenizerReflClass->getProperty('reservedToplevel')->getDefaultValue();
+        /** @var list<string> $tokenizerReservedNewline */
+        $tokenizerReservedNewline = $tokenizerReflClass->getProperty('reservedNewline')->getDefaultValue();
```

Comment:
Instead of painstakingly using reflection to access these 3 properties, you could bind an extra method to the class:

```php
        $assertReservedTopLevelAndNewlineKeywordsAreReserved = Closure::bind(
            function (): void {
                /** @psalm-scope-this Tokenizer */
                $kws = [];

                foreach ([...$this->reservedToplevel, ...$this->reservedNewline] as $v) {
                    foreach (explode(' ', $v) as $kw) {
                        $kws[] = $kw;
                    }
                }

                $kwsDiff = array_diff($kws, $this->reserved);

                TestCase::assertSame([], $kwsDiff);
            },
            new Tokenizer(),
            Tokenizer::class,
        );
        assert($assertReservedTopLevelAndNewlineKeywordsAreReserved instanceof Closure);
        $assertReservedTopLevelAndNewlineKeywordsAreReserved();
```

Reply:
I know `Closure::bind` but as you mentioned in the code example Psalm needs the `@var` comment as well so I definitely do not find it better.

Once https://github.com/phpstan/phpstan/issues/3913#issuecomment-2117363294 will be implemented the current solution should be best and no `@var` will be needed. Psalm does not need `@var` for the current solution already.

Reply:
> but as you mentioned in the code example Psalm needs the @var comment as well so I definitely do not find it better.

What are you talking about? My code sample does not contain any occurrence of `@var`

Reply:
Sorry, I meant "phpdoc comment" - `@psalm-scope-this`.

Reply:
Oh I see… Even with `@var` gone, I think I still prefer my solution:
- no reflection
- fewer variables

Reply:
In #106 the data are even more asserted, so I moved the reflection part into a private getter. Is that fine and can thsi PR be merged?